backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java

```diff
@@ -20,4 +23,18 @@ public static final LocalDate parseUserDate(String userSuppliedDateString) {
 			return LocalDate.parse(userSuppliedDateString); // ISO_LOCAL_DATE is the default
 		}
 	}
+
+	public static String parsePhoneNumber(String userSuppliedPhoneNumber) {
+		if (userSuppliedPhoneNumber == null) {
+			return null;
+		}
+
+		try {
+			var phoneUtil = PhoneNumberUtil.getInstance();
+			return phoneUtil.format(phoneUtil.parse(userSuppliedPhoneNumber, "US"),
+					PhoneNumberUtil.PhoneNumberFormat.NATIONAL);
+		} catch (NumberParseException parseException) {
```

Comment:
The ticket says the problem we have is that only a "phone number" matching the regex of `^[0-9\(\)\- \b]+$` is accepted by the data hub. It seems like this should be a last-ditch thing that we'd hit only if the front end somehow allowed a bad phone number to get through. At the moment there's no validation of phone numbers on the front end but we can add a regex for that.

This change implements much more than just a legal character check, requiring the number to be parseable by `PhoneNumberUtil` and then reformatting it. It [wasn't clear to me](https://javadoc.io/doc/com.googlecode.libphonenumber/libphonenumber/latest/com/google/i18n/phonenumbers/PhoneNumberUtil.html) when the condition "not considered to be a possible number" might occur. The docs do say that it converts vanity numbers (letters to digits) but I wonder if we even want to allow that.

Reply:
This does go farther than what was requested in the ticket, but the motivation given for the ticket is that certain numbers will get rejected by the data hub during a data transfer. I dug into what the data hub is doing and found that [it's using `libphonenumber`, too](https://github.com/CDCgov/prime-data-hub/blob/aaf12bff08c4324df600ae91b8622a399a5c00b8/prime-router/src/main/kotlin/Element.kt#L357). "Not considered to be a possible number" goes beyond what can be enforced in a simple regex and includes cases where the area code or prefix is reserved by NANPA other uses (e.g., in `(aaa) ppp-ssss`, neither the `a` nor `p` blocks can start with `1` or correspond to sequences reserved for other uses like `555`, `311`, etc.).

If we go with a regex approach, we could still have the validation issues described in #340 where an invalid number gets accepted by simple-report and later gets rejected by the data hub.